### PR TITLE
Refactor

### LINKED ISSUE #52: Refactoring PR
Refactors previous PR
- commands/scan.ts
    -  remove isVerbose check for displaying failed tests
- test/azure/rules.spec.ts
    - inline creation of rules to make it easier to read
    - use equal instead of isAtLeast => we expect 0 
- test/azure/scan.spec.ts
    - test formatting of scan command    
        - Let's talk about what to do on this one
        - probably needs to be moved to a unit test
        - how do I test a command method without running it through a cli?
- test/azure/scanner.spec.ts
    - isAtLeast - we should know the exact answers we're looking for in our test cases here
    - More really specific assertions here that should probably be its own test case (or something else)
- test/unit/commands/hello.spec.ts
    - can we delete?
- No tests on throwaway code!

### PR DESCRIPTION
**Pull request by Joshua-Phelps**
closes #52 

- removes verbose check for displaying failed tests
- cleans up rule creation in test/azure/rules.spec.ts
- extracts the scan format method into a helper function so it can be tested
- removes tests for throwaway code
- resourceGraphRule.toScanResult() method is public 
    - adds a test
    - adds check for id column in the ResourcesResponse since its not explicitly guaranteed
- removes hello command and tests

### CODE REVIEW

File: test/unit/rules.spec.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    resultTruncated: 'false',
+    data: {columns: [{name: 'id', type: 'string'}], rows: [['mockResourceId']]},
+    _response: {
+      request: {} as WebResourceLike,
+      status: 200,
+      headers: {} as HttpHeadersLike,
+      bodyAsText: '',
+      parsedBody: {} as ResourceGraphModels.QueryResponse,
+    },
+  };
```
Comment (Joshua-Phelps) [inline test/unit/rules.spec.ts:23]: is there a preferred approach to mocking something like this?

File: src/commandHelper.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+export interface LogOptions {
+  indent?: number;
+  color?: string;
+  bold?: boolean;
+}
+
+export const format = (message: string, options?: LogOptions) => {
+  let formattedMessage = message;
+  if (options?.indent && options.indent > 0) {
```
Comment (noelbundick) [inline src/commandHelper.ts:11]: nit: just wanted to make sure you were aware of [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)

```javascript
if (options?.indent > 0) {
```

File: test/unit/helper.spec.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    };
+    let formattedMessage = format(message, options);
+    expect(formattedMessage).to.equal('  ' + message, 'adds 2 spaces');
+
+    options.indent = -1;
+    formattedMessage = format(message, options);
+    expect(formattedMessage).to.equal(
+      message,
+      'no spaces added if indent number is < 0'
+    );
```
Comment (noelbundick) [inline test/unit/helper.spec.ts:18]: In general, I'd recommend testing for specific conditions like this into their own test case, ex: `it('should not add spaces with a negative indent')` or something

File: test/unit/rules.spec.ts (hunk 1 above)
Comment (noelbundick) [reply test/unit/rules.spec.ts:23]: A const object like this works great. If we need to reuse across files, pulling this out into a `testData.ts` file works as well. There are also tools that can run/record HTTP requests, save to disk, & then replay with that saved data.

File: test/unit/rules.spec.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    );
+    expect(scanResult).to.deep.equal({
+      ruleName: mockRule.name,
+      description: mockRule.description,
+      total: 1,
+      resourceIds: ['mockResourceId'],
+    });
+  });
+  it("should throw an errow if the 'id' column is not returned from Resource Graph", async () => {
+    mockResourcesResponse.data.columns = [];
```
Comment (noelbundick) [inline test/unit/rules.spec.ts:43]: We're mutating the global test data here. If tests run in parallel, and/or if we add more tests, then results become unpredictable. 2 quick ideas

1: create the specific test data you need inside each test case
2: change from a global const to a function that returns a new object every time, and then you can do whatever you want to modify that new object w/o affecting other tests (assuming that function isn't also using globals for its child properties, etc)

File: test/unit/rules.spec.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+      parsedBody: {} as ResourceGraphModels.QueryResponse,
+    },
+  };
+  const mockRule: ResourceGraphRule = {
+    name: 'test-rule',
+    query: 'mock query',
+    description: 'Intentional bad query',
+    type: RuleType.ResourceGraph,
+  };
+  it('can produce a scan result', async () => {
```
Comment (noelbundick) [inline test/unit/rules.spec.ts:30]: nit: either these don't need to be async or we're missing an await in the body

**Review by noelbundick — APPROVED** [review]
lgtm w/ some comments

**Pull request merged**